### ISSUE TITLE
Unable to reindex Product Prices

### ISSUE THREAD

**Issue by vberthet** [magento/magento2#36944]
### Preconditions and environment

- Magento version 2.4.5-p1
- Multi-website configuration

### Steps to reproduce

1. Create a new bundle product
2. In "Product in Websites" unchecked all websites
3. run bin/magento indexer:reindex

### Expected result

Index should be rebuild without errors

### Actual result

Product Price index crash with : 
`Product Price index process error during indexation process: Notice: Undefined offset: 1393 in /home/dexergie/www/releases/173/vendor/magento/module-bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php on line 124 `

### Additional information

Inside Magento\Bundle\Model\ResourceModel\Indexer\Price\DisabledProductOptionPriceModifier :
```php
private function getWebsiteIdsOfProduct(int $entityId): array
    {
        if (isset($this->websiteIdsOfProduct[$entityId])) {
            return $this->websiteIdsOfProduct[$entityId];
        }
        $connection = $this->resourceConnection->getConnection('indexer');
        $select = $connection->select();
        $select->from(
            ['product_in_websites' => $this->resourceConnection->getTableName('catalog_product_website')],
            ['website_id']
        )->where('product_in_websites.product_id = ?', $entityId);
        foreach ($connection->fetchCol($select) as $websiteId) {
            $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
        }
        return $this->websiteIdsOfProduct[$entityId];
    }
```

If the product as not website assigned $connection->fetchCol($select) will return an empty array so $this->websiteIdsOfProduct[$entityId] will never be populated. 

This could be fixed by : 
```php
private function getWebsiteIdsOfProduct(int $entityId): array
    {
        if (isset($this->websiteIdsOfProduct[$entityId])) {
            return $this->websiteIdsOfProduct[$entityId];
        }
        $connection = $this->resourceConnection->getConnection('indexer');
        $select = $connection->select();
        $select->from(
            ['product_in_websites' => $this->resourceConnection->getTableName('catalog_product_website')],
            ['website_id']
        )->where('product_in_websites.product_id = ?', $entityId);
        $this->websiteIdsOfProduct[$entityId] = [];
        foreach ($connection->fetchCol($select) as $websiteId) {
            $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
        }
        return $this->websiteIdsOfProduct[$entityId];
    }
```

See Issue [36139](https://github.com/magento/magento2/issues/36139) for previous repport

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @vberthet ,

Verified the issue in 2.4-develop instance and its not reproducible,We are getting expected result as per the description.

Preconditions:
Magento version 2.4-develop
PHP Version *8.1

Steps to reproduce:
1.Install Magento instance
2.Create Multi Website in Backend
3.Create Bundle Product
4.Uncheck all the website in the Bundle Product
5.Run Reindexer Command

Screenshots:
<img width="1392" alt="web1" src="https://user-images.githubusercontent.com/60198592/222617680-823f38e0-1c3b-4596-895a-531314a4e9f3.png">
<img width="1392" alt="web2" src="https://user-images.githubusercontent.com/60198592/222617699-62c8c009-12b6-4632-bb27-13953f508022.png">

Reindexer command run successfully without any issue.Kindly let me know more information if the issue is still reproducible.

Let us know if we missing any Steps

Regards,

**Comment by vberthet** [comment]
@magento give me 2.4-develop instance

**Comment by vberthet** [comment]
Fixed in 2.4-develop by commit https://github.com/magento/magento2/commit/0d90748aa1dd76581b8a3dbf34b5ce82961d9902
